src/main/oAuthClient.ts

```diff
@@ -276,6 +286,52 @@ export default class OAuthClient {
     })
   }
 
+  loginWithIdToken(provider: string, idToken: string, nonce: string, opts: AuthOptions = {}): Promise<void> {
+    const authParams = this.authParams({
+      ...opts,
+      useWebMessage: false
+    })
+
+    return this.loginWithRedirect({
+      provider,
+      idToken,
+      nonce,
+      ...authParams
```

Comment:
`nonce` is already defined in `opts`, which override the ones passed in parameter.